hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/GrpcReplicationClient.java

```diff
@@ -161,6 +161,9 @@ public void onNext(CopyContainerResponseProto chunk) {
       try {
         chunk.getData().writeTo(stream);
       } catch (IOException e) {
+        LOG.error("Failed to write the stream buffer to {} for container {}",
+            outputPath, containerId, e);
+        deleteOutputOnFailure();
```

Comment:
```suggestion
onError(e);
```

Reply:
IMO, the original LOG msg is better, because by looking at the JIRA descriptions,  actually the download via network is successful, the failure happens at sync the downloaded data into the on-disk file.
So we don't call onError() here which should be called on network transfer faliures, and it may not be a good style to call an overrided method in another(onError in onNext here).

Reply:
May be we could extract common error handling stuff into a new method.

Reply:
> May be we could extract common error handling stuff into a new method.

looks good. i agree

Reply:
stream.close();  should be called together with deleteOutputOnFailure();